src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1081,12 +1081,62 @@ class CreateRoleStatementSegment(ansi.CreateRoleStatementSegment):
         insert=[
             Sequence(
                 Ref.keyword("WITH", optional=True),
-                # Very permissive for now. Anything can go here.
-                Anything(),
+                AnySetOf(
+                    OneOf("SUPERUSER", "NOSUPERUSER"),
+                    OneOf("CREATEDB", "NOCREATEDB"),
+                    OneOf("CREATEROLE", "NOCREATEROLE"),
+                    OneOf("INHERIT", "NOINHERIT"),
+                    OneOf("LOGIN", "NOLOGIN"),
+                    OneOf("REPLICATION", "NOREPLICATION"),
+                    OneOf("BYPASSRLS", "NOBYPASSRLS"),
+                    Sequence("CONNECTION", "LIMIT", Ref("NumericLiteralSegment")),
+                    Sequence("PASSWORD", OneOf(Ref("QuotedLiteralSegment"), "NULL")),
+                    Sequence("VALID", "UNTIL", Ref("QuotedLiteralSegment")),
+                    Sequence("IN", "ROLE", Ref("RoleReferenceSegment")),
+                    Sequence("IN", "GROUP", Ref("RoleReferenceSegment")),
+                    Sequence("ROLE", Ref("RoleReferenceSegment")),
+                    Sequence("ADMIN", Ref("RoleReferenceSegment")),
+                    Sequence("USER", Ref("RoleReferenceSegment")),
+                    Sequence("SYSID", Ref("NumericLiteralSegment")),
+                ),
             )
         ],
     )
 
+class CreateUserStatementSegment(BaseSegment):
+    """A `CREATE USER` statement.
+
+    As per:
+    https://www.postgresql.org/docs/current/sql-createuser.html
+    """
+
+    type = "create_user"
+    match_grammar = Sequence(
+        "CREATE",
+        "USER",
+        Ref("ObjectReferenceSegment"),
+        Sequence(
+            Ref.keyword("WITH", optional=True),
+            AnySetOf(
+                OneOf("SUPERUSER", "NOSUPERUSER"),
+                OneOf("CREATEDB", "NOCREATEDB"),
+                OneOf("CREATEROLE", "NOCREATEROLE"),
+                OneOf("INHERIT", "NOINHERIT"),
+                OneOf("LOGIN", "NOLOGIN"),
+                OneOf("REPLICATION", "NOREPLICATION"),
+                OneOf("BYPASSRLS", "NOBYPASSRLS"),
+                Sequence("CONNECTION", "LIMIT", Ref("NumericLiteralSegment")),
+                Sequence("PASSWORD", OneOf(Ref("QuotedLiteralSegment"), "NULL")),
+                Sequence("VALID", "UNTIL", Ref("QuotedLiteralSegment")),
+                Sequence("IN", "ROLE", Ref("RoleReferenceSegment")),
+                Sequence("IN", "GROUP", Ref("RoleReferenceSegment")),
+                Sequence("ROLE", Ref("RoleReferenceSegment")),
+                Sequence("ADMIN", Ref("RoleReferenceSegment")),
+                Sequence("USER", Ref("RoleReferenceSegment")),
+                Sequence("SYSID", Ref("NumericLiteralSegment")),
+            ),
+        ),
+    )
```

Comment:
If this is identical to above you should be able to do this, to reduce code duplicate.

```suggestion
    match_grammar = postgres.CreateRoleStatementSegment
```

Reply:
It's unfortunately not 100% identical as the ANSI definition is "CREATE" "ROLE" for the first two lines and this is "CREATE" "USER". Is there a way for me to merge them or do a partial override?

Reply:
We could just have `CREATE USER` be part of `CREATE ROLE`?

Or could try to do an `insert` and a `remove`, though not 100% sure it will work:
```py
match_grammar = postgres.CreateRoleStatementSegment.match_grammar.copy(
    insert=[Ref.keyword("USER")],
    before=Ref.keyword("ROLE"),
    remove=[Ref.keyword("ROLE")],
)
```

Or we could create a `UserRoleGrammar` that both can refer to?

WDYT?

Reply:
I like the idea of combining the two, postgres docs make it clear that USER is just an alias of ROLE in how they handle it so would make sense to do the same treatment here. Let me give that a pass.